src/Symfony/Component/DependencyInjection/ContainerBuilder.php

```diff
@@ -545,6 +545,9 @@ public function compile()
             if (!$definition->isPublic()) {
                 $this->privates[$id] = true;
             }
+            if ($definition->isSynthetic()) {
+                $this->synthetics[$id] = true;
```

Comment:
Agree. I was also thinking `$this->serviceTypes` something to track&check things in 1 go.

`$this->serviceTypes[$id] & self::IS_PRIVATE`